File: app/src/main/java/antoninovitale/dropcodechallenge/details/BeerDetailsView.java
package antoninovitale.dropcodechallenge.details;

import java.util.List;

import antoninovitale.dropcodechallenge.details.section.model.HeaderSectionModel;
import antoninovitale.dropcodechallenge.details.section.model.IngredientSectionModel;
import antoninovitale.dropcodechallenge.details.section.model.MethodSectionModel;
import antoninovitale.dropcodechallenge.details.section.model.Status;

/**
 * Created by antoninovitale on 02/09/2017.
 */
interface BeerDetailsView {
    void setupHeaderSection(HeaderSectionModel headerSectionModel);

    void setupMaltsSection(List<IngredientSectionModel> malts);

    void setupHopsSection(List<IngredientSectionModel> hops);

    void setupMethodSection(List<MethodSectionModel> methods);

    void notifyMaltDone(int position);

    void notifyHopDone(int position);

    void notifyMethodStatusChanged(int position, Status status);
}

File: app/src/main/java/com/ninovitale/punkapi/app/details/BeerDetailsView.kt
package com.ninovitale.punkapi.app.details

import com.ninovitale.punkapi.app.details.section.model.HeaderSectionModel
import com.ninovitale.punkapi.app.details.section.model.IngredientSectionModel
import com.ninovitale.punkapi.app.details.section.model.MethodSectionModel
import com.ninovitale.punkapi.app.details.section.model.Status

/**
 * Created by antoninovitale on 02/09/2017.
 */
interface BeerDetailsView {
    fun setupHeaderSection(headerSectionModel: HeaderSectionModel)
    fun setupMaltsSection(malts: List<IngredientSectionModel>)
    fun setupHopsSection(hops: List<IngredientSectionModel>)
    fun setupMethodSection(methods: List<MethodSectionModel>)
    fun notifyMaltDone(positionInSection: Int, globalPosition: Int)
    fun notifyHopDone(positionInSection: Int, globalPosition: Int)
    fun notifyMethodStatusChanged(positionInSection: Int, globalPosition: Int, status: Status)
    fun clearSections()
}